velox/type/tests/DecimalTest.cpp

```diff
@@ -287,84 +291,198 @@ TEST(DecimalAggregateTest, adjustSumForOverflow) {
 }
 
 TEST(DecimalTest, rescaleDouble) {
-  assertRescaleDouble<int64_t>(-3333.03, DECIMAL(10, 4), -33'330'300);
-  assertRescaleDouble<int128_t>(-3333.03, DECIMAL(20, 1), -33'330);
-  assertRescaleDouble<int128_t>(-3333.03, DECIMAL(20, 10), -33'330'300'000'000);
-
-  assertRescaleDouble<int64_t>(-2222.02, DECIMAL(10, 4), -22'220'200);
-  assertRescaleDouble<int128_t>(-2222.02, DECIMAL(20, 1), -22'220);
-  assertRescaleDouble<int128_t>(-2222.02, DECIMAL(20, 10), -22'220'200'000'000);
-
-  assertRescaleDouble<int64_t>(-1.0, DECIMAL(10, 4), -10'000);
-  assertRescaleDouble<int128_t>(-1.0, DECIMAL(20, 1), -10);
-  assertRescaleDouble<int128_t>(-1.0, DECIMAL(20, 10), -10'000'000'000);
-
-  assertRescaleDouble<int64_t>(0.00, DECIMAL(10, 4), 0);
-  assertRescaleDouble<int128_t>(0.00, DECIMAL(20, 1), 0);
-  assertRescaleDouble<int128_t>(0.00, DECIMAL(20, 10), 0);
-
-  assertRescaleDouble<int64_t>(100, DECIMAL(10, 4), 1'000'000);
-  assertRescaleDouble<int128_t>(100, DECIMAL(20, 1), 1'000);
-  assertRescaleDouble<int128_t>(100, DECIMAL(20, 10), 1'000'000'000'000);
-
-  assertRescaleDouble<int64_t>(99999.99, DECIMAL(10, 4), 999'999'900);
-  assertRescaleDouble<int128_t>(99999.99, DECIMAL(20, 1), 1'000'000);
-  assertRescaleDouble<int128_t>(99999.99, DECIMAL(20, 10), 999'999'900'000'000);
-
-  assertRescaleDouble<int128_t>(
+  assertRescaleFloatingPoint<double, int64_t>(
+      -3333.03, DECIMAL(10, 4), -33'330'300);
+  assertRescaleFloatingPoint<double, int128_t>(
+      -3333.03, DECIMAL(20, 1), -33'330);
+  assertRescaleFloatingPoint<double, int128_t>(
+      -3333.03,
+      DECIMAL(38, 18),
+      HugeInt::build(0xFFFFFFFFFFFFFF4B, 0x50EABA2657C90000));
```

Comment:
This is not very readable. Is there a way to write these tests using human-readable values? 

'-333303000000000000000'

Reply:
Updated to use HugeInt::parse to make them more readable. Thanks.